addons/html_builder/static/src/website_builder/plugins/options/google_map_option/google_map_option_plugin.js

```diff
@@ -0,0 +1,333 @@
+     * @returns {Promise<ApiKeyValidation>}
+     */
+    async validateGMapApiKey(key) {
+        if (key) {
+            try {
+                const response = await fetch(`https://maps.googleapis.com/maps/api/staticmap?center=belgium&size=10x10&key=${encodeURIComponent(key)}`);
+                const isValid = (response.status === 200);
+                return {
+                    isValid,
+                    message: isValid
+                        ? undefined
+                        : _t("Invalid API Key. The following error was returned by Google: %(error)s", { error: await response.text() }),
+                };
+            } catch {
+                return {
+                    isValid: false,
+                    message: _t("Check your connection and try again"),
+                };
+            }
+        } else {
+            return { isValid: false };
+        }
+    }
+
+    /**
+     * @param {Element} editingElement
+     * @param {Coordinates} coordinates
+     * @param {boolean} [notify=true]
+     * @returns {Promise<Place|undefined>}
+     */
+    async nearbySearch(editingElement, coordinates, notify = true) {
+        const place = this.gpsMapCache.get(coordinates);
+        if (place) {
+            return place;
+        }
+
+        const p = coordinates.substring(1).slice(0, -1).split(',');
+        const location = new this.mapsAPI.LatLng(p[0] || 0, p[1] || 0);
+        return new Promise(resolve => {
+            const placesService = new this.placesAPI.PlacesService(document.createElement('div'));
+            placesService.nearbySearch({
+                // Do a 'nearbySearch' followed by 'getDetails' to avoid using
+                // GMap Geocoder which the user may not have enabled... but
+                // ideally Geocoder should be used to get the exact location at
+                // those coordinates and to limit billing query count.
+                location,
+                radius: 1,
+            }, (results, status) => {
+                const GMAP_CRITICAL_ERRORS = [
+                    this.placesAPI.PlacesServiceStatus.REQUEST_DENIED,
+                    this.placesAPI.PlacesServiceStatus.UNKNOWN_ERROR
+                ];
+                if (status === this.placesAPI.PlacesServiceStatus.OK) {
+                    placesService.getDetails({
+                        placeId: results[0].place_id,
+                        fields: ['geometry', 'formatted_address'],
+                    }, (place, status) => {
+                        if (status === this.placesAPI.PlacesServiceStatus.OK) {
+                            this.gpsMapCache.set(coordinates, place);
+                            resolve(place);
+                        } else if (GMAP_CRITICAL_ERRORS.includes(status)) {
+                            if (notify) {
+                                this.notifyGMapError(editingElement);
+                            }
+                            resolve();
+                        }
```

Comment:
what happens in the else ? you want to never resolve ?
why not resolve at the end of the function by default

Reply:
This was basically copy-pastes from website. I made a separate commit to change that implementation though because I also thought it was faulty. There were lots of weird things with promises. I did it separately because I figured I was not supposed to change their implementation if not strictly necessary for the conversion. I missed this `else` case though, I'm going to add it.